src/components/transport_manager/src/tcp/tcp_device.cc

```diff
@@ -48,9 +48,14 @@ TcpDevice::TcpDevice(const utils::HostAddress& address, const std::string& name)
 }
 
 bool TcpDevice::IsSameAs(const Device* other) const {
-  SDL_AUTO_TRACE();
-  SDL_DEBUG("Device: " << other);
-  const TcpDevice* other_tcp_device = static_cast<const TcpDevice*>(other);
+  SDL_TRACE("enter. Device: " << other->unique_device_id());
+
+  // TODO(AKutsan): remove dynamic cast after APPLINK-25700
```

Comment:
@LuxoftAKutsan, how someone who will make APPLINK-25700 will know that it should be deleted?